iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -1038,6 +1050,28 @@ static String encodeString(String rawString) {
     return HiveConf.EncoderDecoderFactory.URL_ENCODER_DECODER.encode(rawString);
   }
 
+  static String getLocation(String location, String defaultLocation, Configuration conf) {
+    if (location == null || defaultLocation == null) {
+      return location;
+    }
+    boolean maskDefaultLocation = conf.getBoolean(HiveConf.ConfVars.HIVE_ICEBERG_MASK_DEFAULT_LOCATION.varname,
+        HiveConf.ConfVars.HIVE_ICEBERG_MASK_DEFAULT_LOCATION.defaultBoolVal);
+    try {
+      return maskDefaultLocation ?
+          getQualifiedPath(location, conf).replaceFirst(getQualifiedPath(defaultLocation, conf),
+              TABLE_DEFAULT_LOCATION) :
+          location;
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  static String getQualifiedPath(String location, Configuration conf) throws IOException {
+    Path path = new Path(location);
+    FileSystem fs = path.getFileSystem(conf);
```

Comment:
Maybe it is not true for java. But this type of resource should be closable (disposable) in .net. Aka it should be close the resource. I guess there could be some mechanism in Java that could close automatically, but even if it has (don't know) but the best practice is always to close a resource that is not used anymore. Especially those that is coming from the OS

Reply:
FileSystem is cached, it isn't created always, it is created once & next time when you try to fetch it, it returns the same instance not a new one. In the end, all the FS instances are closed for the UGI which were created by Hive during the query execution, so there isn't no explicit need for closing the FS, it will get closed in the end.
If we close it in the middle, similar issues will arise like in: [HIVE-27447](https://issues.apache.org/jira/browse/HIVE-27447)

Reply:
Ack